agent/main.go

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright © 2024 Pulumi Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import "github.com/pulumi/pulumi-kubernetes-operator/agent/cmd"
+
+func main() {
+	cmd.Execute()
+}
```

Comment:
Structure nit: this is probably more at home under `cmd` since that's where binaries typically live. Something like this
```
v2
├── cmd
│   ├── agent
│   │   ├── init.go
│   │   ├── main.go
│   │   ├── root.go
│   │   └── serve.go
│   └── operator
│       └── main.go
├── go.mod (optional -- "module github.com/pulumi/pulumi-kubernetes-operator/v2")
├── internal
│   ├── operataor
│   └── server
└── proto
```

Note also the absence of `pkg` -- "a package to hold other packages" doesn't really serve a purpose and is considered an anti-pattern. On the other hand, `internal` can effectively accomplish the same while also allowing you to not worry about public APIs until they actually matter (which can be never).